rust/lance-core/src/utils/path.rs

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use object_store::path::Path;
+
+pub trait LancePathExt {
+    fn child_path(&self, path: &Path) -> Path;
```

Comment:
I couldn't figure out how to join two `object_store::Path`.  If I missed something let me know.

Reply:
I don't think there is.